docs/running_neps/configuring_runs.md

```diff
@@ -0,0 +1,123 @@
+# Configuring and Running Optimizations
+
+The `neps.run` function is the core interface for running Hyperparameter and/or architecture search using optimizers in NePS.
+
+This document breaks down the core arguments that allow users to control the optimization process in NePS. 
+
+## Search Strategy
+At default NePS intelligently selects the most appropriate search strategy based on your defined configurations in
+`pipeline_space`.
+The characteristics of your search space, as represented in the `pipeline_space`, play a crucial role in determining
+which optimizer NePS will choose. This automatic selection process ensures that the strategy aligns perfectly
+with the specific requirements and nuances of your search space, thereby optimizing the effectiveness of the
+hyperparameter and/or architecture optimization. You can also manually select a specific or custom optimizer that better
+matches your specific needs. For more information, refer [here](https://automl.github.io/neps/latest/optimizers).
+
+## Arguments
+
+
+## Setup `run_args`
+
+To configure the execution parameters, you can also define them via a YAML file provided through the `run_args`
+argument. Parameters not explicitly defined within this file will receive their default values. Below is
+an example of how to structure your YAML configuration.
+
+```yaml
+run_args:  # Essential starting key; everything under this key gets considered for configuration.
+  run_pipeline:
+    path: "path/to/your/run_pipeline.py" # File path of the run_pipeline function
```

Comment:
can all this be instead extracted from the docstrings?
that alleviates the need for separate doc maintenance and also enforces the developer to be descriptive with the docstrings
I tried this with `mkdocstrings` but could not resolve some dependency issues 
In case this is not solvable quickly then no need to add this to the current PR but a new issue documenting this would be nice and we look into it after our release.

Reply:
I agree, for now just leave it and it can be re-worked. The more documentation we can put in the code itself, the better. I have some experience on doing this so feel free to raise an issue that this needs to be fixed before a release.

Reply:
Also never duplicate documentation if possible. Reason being is everyone is already bad enough at keeping documentation up to date with code, having multiple places it needs to be updated becomes a maintanence hazard.

Reply:
I raised an issue about this.